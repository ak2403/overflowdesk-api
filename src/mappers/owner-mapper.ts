import { Mapper } from "../types/mappers/mapper";
import { Owner } from "../types/mappers/owner";

export class OwnerMapper implements Mapper<Owner> {
  private _accountId!: number;
  private _name!: string;
  private _profileImage!: string;
  private _profileLink!: string;
  private _reputation!: number;
  private _userId!: number;

  set accountId(id: number) {
    this._accountId = id;
  }

  set name(name: string) {
    this._name = name;
  }

  set profileImage(linkToImage: string) {
    this._profileImage = linkToImage;
  }

  set profileLink(linkToProfile: string) {
    this._profileLink = linkToProfile;
  }

  set reputation(reputation: number) {
    this._reputation = reputation;
  }

  set userId(id: number) {
    this._userId = id;
  }

  build(): Owner {
    return {
      id: this._accountId,
      name: this._name,
      profileImage: this._profileImage,
      profileLink: this._profileLink,
      reputation: this._reputation,
      userId: this._userId,
    };
  }
}
