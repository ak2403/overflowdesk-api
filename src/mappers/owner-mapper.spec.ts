import { OwnerMapper } from "./owner-mapper";

describe("OwnerMapper()", () => {
  const ownerMapper = new OwnerMapper();

  it("construct owner for passed attribute values", () => {
    ownerMapper.accountId = 123;
    ownerMapper.userId = 456;
    ownerMapper.reputation = 1;
    ownerMapper.profileImage = "sample-link";

    expect(ownerMapper.build()).toMatchSnapshot();
  });
});
