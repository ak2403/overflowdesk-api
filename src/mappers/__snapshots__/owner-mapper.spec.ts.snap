// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnerMapper() construct owner for passed attribute values 1`] = `
{
  "id": 123,
  "name": undefined,
  "profileImage": "sample-link",
  "profileLink": undefined,
  "reputation": 1,
  "userId": 456,
}
`;
